staging/src/k8s.io/kubectl/pkg/cmd/exec/exec.go

```diff
@@ -120,10 +120,18 @@ type RemoteExecutor interface {
 type DefaultRemoteExecutor struct{}
 
 func (*DefaultRemoteExecutor) Execute(method string, url *url.URL, config *restclient.Config, stdin io.Reader, stdout, stderr io.Writer, tty bool, terminalSizeQueue remotecommand.TerminalSizeQueue) error {
+	// Legacy SPDY executor is default. If feature gate enabled, fallback
+	// executor attempts websockets first--then SPDY.
 	exec, err := remotecommand.NewSPDYExecutor(config, method, url)
 	if err != nil {
 		return err
 	}
+	if cmdutil.RemoteCommandWebsockets.IsEnabled() {
+		exec, err = remotecommand.NewFallbackExecutor(config, method, url)
```

Comment:
once the fallback executor becomes generic, this would look like:

```go
websocketExec, err := remoteCommand.NewWebsocketExecutor(...)
if err != nil {
  return err
}
exec, err = remoteCommand.NewFallbackExecutor(websocketExec, exec, isUpgradeFailureError)
...
```

Reply:
Fixed.